w3-hw1-ThuyDung/cypress/e2e/formPage.cy.js

```diff
@@ -0,0 +1,274 @@
+import { formPage } from "../pages/formPage";
+
+describe('User can register', () => {
+  beforeEach(() => {
+    cy.fixture("data").as("data");
+    cy.visit(Cypress.env("form"));
+  });
+  it("TC-FORM-05: Verify that user can registers when user inputs all fields with valid data ", () => {
+    cy.get("@data").then((data) => {
+      formPage
+        .typeFirstName(data.profiles[5].firstname)
+        .typeLastName(data.profiles[5].lastname)
+        .typeEmail(data.profiles[5].email)
+        .selectGender(data.profiles[5].gender)
+        .typeMobile(data.profiles[5].mobile)
+        .selectDOB(data.profiles[5].DOB.day, data.profiles[5].DOB.month, data.profiles[5].DOB.year)
+        .selectSubjects(data.profiles[5].subjects)
+        .checkHobbies(data.profiles[5].hobbies)
+        .uploadPicture(data.profiles[5].picture)
+        .typeCurrentAddress(data.profiles[5].currentaddress)
+        .selectStateAndCity(data.profiles[5].state, data.profiles[5].city)
+        .clickSubmit();
+      formPage.popup
+        .isNameCorret(data.profiles[5].firstname, data.profiles[5].lastname)
+        .isEmaiCorret(data.profiles[5].email)
+        .isGenderCorret(data.profiles[5].gender)
+        .isMobileCorret(data.profiles[5].mobile)
+        .isDOBCorret(data.profiles[5].DOB.day,data.profiles[5].DOB.month,data.profiles[5].DOB.year)
+        .isSubjectsCorrect(data.profiles[5].subjects)
+        .isHobbiesCorrect(data.profiles[5].hobbies)
+        .isPictureCorrect(data.profiles[5].picture)
+        .isCurrentAddressCorrect(data.profiles[5].currentaddress)
+        .isStateAndCityCorrect(data.profiles[5].state, data.profiles[5].city)
+    });
+  });
+  it("TC-FORM-06: Verify that user can register when user inputs mandatory fields with valid data", ()=>{
+    cy.get("@data").then((data) => {
+      formPage
+        .typeFirstName(data.profiles[6].firstname)
+        .typeLastName(data.profiles[6].lastname)
+        .selectGender(data.profiles[6].gender)
+        .typeMobile(data.profiles[6].mobile)
+        .clickSubmit();
+      formPage.popup
+        .isNameCorret(data.profiles[6].firstname, data.profiles[6].lastname)
+        .isGenderCorret(data.profiles[6].gender)
+        .isMobileCorret(data.profiles[6].mobile)
+    });
+  });
+});
+describe('User can not register',()=>{
+  beforeEach(() => {
+    cy.fixture("data").as("data");
+    // cy.visit(Cypress.env("form"));
+    cy.visit(Cypress.env("form"), {
+      timeout: 1000000 
+    });
+  });
+  it("TC-FORM-07: Verify that user cannot register when user doesn't input any mandatory fields",()=>{
+    cy.get("@data").then((data) =>{
+      formPage
+        .clickSubmit()
+        .isFirtsNameValid(false)
+        .isLastNameValid(false)
+        .isGenderValid(false)
+        .isMobileValid(false);
+    })
+  });
+  it("TC-FORM-10: Verify that user cannot register when inputting only blank spaces on First Name field", () =>{
```

Comment:
This is for all unhappy cases, you cannot do them all case-by-case because it will take much effort. Combine them, ex: `Submit fail with invalid email`, your test data will look like this:
```json
{
  "email": {
    "valid": ["abc@gmail.com", "a12@yahoo.com"],
    "invalid": ["@@@", "abc...."]
  }
}
```
Then you loop all the invalid data because just the same output for all invalid cases